pkg/operator/controller/dns/controller.go

```diff
@@ -352,6 +361,24 @@ func (r *reconciler) publishRecordToZones(zones []configv1.DNSZone, record *iov1
 			}
 		} else if isRecordPublished {
 			condition, err = r.replacePublishedRecord(zones[i], record)
+		} else if isDomainPublished, err = domainIsAlreadyPublishedInZone(context.Background(), r.cache, record, &zones[i]); err != nil {
```

Comment:
You don't need to declare `isDomainPublished` outside the `else if` clause.  Better to keep `isDomainPublished` and `err` scoped to the `else if` clauses.  
```suggestion
		} else if isDomainPublished, err := domainIsAlreadyPublishedInZone(context.Background(), r.cache, record, &zones[i]); err != nil {
```